### ISSUE TITLE
Promote download above other links

### ISSUE THREAD

**Issue by quozl** [sugarlabs/aslo-v4#25]
On the activity detail view, the link to source code is offered as a priority above that of download.  An elementary school child may download source code by mistake.  Prioritise the download of the bundle.

**Comment by srevinsaju** [comment]
Needs more review / information. This issue has been discussed similarly in many other places. 
Link to source code, only redirects to GitHub page, an elementary school child may not understand the working of GitHub. I personally feel, that the download button is best placed at the bottom.

**Comment by free-libre-software** [comment]
Download button can be bumped above gallery. Or layout can be redesigned so that source code button is not as big as download button. It can still be confusing even after placing download button above source code button. From UX perspective, download button should be placed somewhere at the top where it is visible on all activity pages without needing to scroll. This is because download will the most frequent action taken by users. So it should require minimal effort (scrolling, looking).

Unrelated to this issue:
On Activity pages, changelog inside `<pre class="pre-scrollable">` tag has CSS property `overflow-y: scroll`. This shows scrollbar on Firefox and Chromium even when text is not scrollable (within max-height limit). It should be either `overflow-y: auto` or not at all. Since it is already inside `<pre>` tag, it should be scrollable both vertically and horizontally anyway without any responsiveness issue.
However, Browse activity (WebKit) renders it fine so this issue can be ignored if do not wish to change or remove bootstrap code from this part.

**Comment by srevinsaju** [comment]
Thanks @free-libre-software, I am not an expert at UX design. I would be grateful if you do
* rearrange the bundle page, to make it look right. The code template is right over here: https://github.com/sugarlabs-appstore/aslo-v4/blob/master/aslo4/constants.py#L88
* or create a UX rendering of the webpage, so I can replicate. Preferably an svg rendering

**Comment by quozl** [comment]
Is it as simple as changing the order of elements?  https://github.com/sugarlabs-appstore/aslo-v4/commit/e927a617e2417bd7d38af09e1c60881bc6fd739d

Could also remove the source code download.  It serves no purpose for the average child user.

**Comment by srevinsaju** [comment]
@quozl does this look right?
![image](https://user-images.githubusercontent.com/48695438/92236656-81f90900-eebe-11ea-8e34-adb5493bc9f4.png)

**Comment by quozl** [comment]
Sorry, I'm confused.

* my question illustrative commit https://github.com/sugarlabs-appstore/aslo-v4/commit/e927a617e2417bd7d38af09e1c60881bc6fd739d you've made into a pull request https://github.com/sugarlabs-appstore/aslo-v4/pull/39, which says "fixes" and "possible fixes", and is directed at my temporary branch,
* you're suggesting the download link is to be below the version and license, both here and in https://github.com/sugarlabs-appstore/aslo-v4/pull/40, which is also directed at my temporary branch,
* there's no answer to my question above https://github.com/sugarlabs-appstore/aslo-v4/issues/25#issuecomment-675281590.

My opinion is that the download link should be between the activity title and version.

**Comment by srevinsaju** [comment]
> Sorry, I'm confused.
> 
> * my question illustrative commit [e927a61](https://github.com/sugarlabs-appstore/aslo-v4/commit/e927a617e2417bd7d38af09e1c60881bc6fd739d) you've made into a pull request #39, which says "fixes" and "possible fixes", and is directed at my temporary branch,
> * you're suggesting the download link is to be below the version and license, both here and in #40, which is also directed at my temporary branch,
> * there's no answer to my question above [#25 (comment)](https://github.com/sugarlabs-appstore/aslo-v4/issues/25#issuecomment-675281590).

What do you mean? Are you asking if its as simple as doing https://github.com/sugarlabs-appstore/aslo-v4/commit/e927a617e2417bd7d38af09e1c60881bc6fd739d or do we have to do something more complex?
Well, I do not know want to move the download button to a new position, but it was your suggestion. So it was a question I asked you, as to where you would like to see the download button. By reading the comment, I inferred if you were asking me "Isn't it simple to do this? if yes, why not move it?", so I created a pull request. 

> My opinion is that the download link should be between the activity title and version.

Thanks. I will update the PR accordingly and merge

**Comment by quozl** [comment]
> What do you mean? Are you asking if its as simple as doing [e927a61](https://github.com/sugarlabs-appstore/aslo-v4/commit/e927a617e2417bd7d38af09e1c60881bc6fd739d) or do we have to do something more complex?

Yes.

> Well, I do not know want to move the download button to a new position, but it was your suggestion.

I can't parse this.

> So it was a question I asked you, as to where you would like to see the download button. By reading the comment, I inferred if you were asking me "Isn't it simple to do this? if yes, why not move it?", so I created a pull request.
> 
> > My opinion is that the download link should be between the activity title and version.
> 
> Thanks. I will update the PR accordingly and merge

Thanks.

**Comment by srevinsaju** [comment]
> > What do you mean? Are you asking if its as simple as doing [e927a61](https://github.com/sugarlabs-appstore/aslo-v4/commit/e927a617e2417bd7d38af09e1c60881bc6fd739d) or do we have to do something more complex?
> 
> Yes.
> 
> > Well, I do not know want to move the download button to a new position, but it was your suggestion.
> 
> I can't parse this.
> 

I meant, I had no problem with the current position where the download button was placed. :smile:, it should be solved on merging the PR mentioned before.

> > So it was a question I asked you, as to where you would like to see the download button. By reading the comment, I inferred if you were asking me "Isn't it simple to do this? if yes, why not move it?", so I created a pull request.
> > 
> > > My opinion is that the download link should be between the activity title and version.
> > 
> > Thanks. I will update the PR accordingly and merge
> 
> Thanks.

**Comment by quozl** [comment]
> I meant, I had no problem with the current position where the download button was placed. 

Yes, I know, because that's where you put it in the first place?  :grin:  What I'm asking is for you to imagine being an elementary school child, in disadvantaged circumstances, trying to figure out what to do to install an activity you've heard about from other children.  What guidance can we give such a child to ensure they choose the right action?  Is a link to source code reasonable; what will be the outcome.  Is the position of the link to download the bundle reasonable; what will be the outcome if not?  Can we expect them to know what a flatpak is, and so on.

> it should be solved on merging the PR mentioned before.

Good, thanks.  I presume you mean https://github.com/sugarlabs-appstore/aslo-v4/pull/39.  I don't know what it looks like.

**Comment by srevinsaju** [comment]
Great, maybe I can put a screenshot. hold on

**Comment by srevinsaju** [comment]
![image](https://user-images.githubusercontent.com/48695438/93167099-8b893900-f728-11ea-80ce-61a831148369.png)

**Comment by srevinsaju** [comment]
@quozl can I merge #39 ?

**Comment by quozl** [comment]
Thanks.  I agree it fixes https://github.com/sugarlabs-appstore/aslo-v4/issues/25.

**Comment by quozl** [comment]
Not fixed here; https://people.sugarlabs.org/~srevinsaju/aslo/app/com.jotaro.ImplodeActivity.html

**Comment by srevinsaju** [comment]
@quozl just fixed it on e5abcce3f59508af596d5a6928f5979295804531 and rebuilt the aslo. Can you please recheck the link?
